### PR TITLE
archi: init at 4.7.1

### PR DESCRIPTION
**Pull request by earldouglas**
###### Motivation for this change

Add a derivation for Archi.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/Archi/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+       *   sha256 = "0zg0bs9i2kkr11bbssfxhrp4ym2zzhb2bk7m38z9cj99h7pmgyhr";
+       * }
+       */
+    else
+       throw "Unsupported system";
+
+  installPhase = ''
+    cp -r . $out
+    mkdir -p $out/bin
+    ln -s $out/Archi $out/bin/Archi
```
Comment (earldouglas) [inline pkgs/tools/misc/Archi/default.nix:30]: This is creating an invalid relative symlink to `../Archi` under NixOS, but it works correctly on Nix+Linux.

**Comment by earldouglas** [comment]
This isn't quite working yet; looks like the binary might need to be patched for NixOS paths.

File: pkgs/tools/misc/Archi/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{ stdenv, fetchurl }:
+
+let version = "4.7.1";
+in stdenv.mkDerivation {
+
+  pname = "Archi";
+  inherit version;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/Archi/default.nix:7]:
```suggestion
stdenv.mkDerivation rec {

  pname = "Archi";
  version = "4.7.1";
```

File: pkgs/tools/misc/Archi/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  src =
+    if stdenv.hostPlatform.system == "x86_64-linux" then
+      fetchurl {
+        url = "https://www.archimatetool.com/downloads/archi/Archi-Linux64-${version}.tgz";
+        sha256 = "0sd57cfnh5q2p17sd86c8wgmqyipg29rz6iaa5brq8mwn8ps2fdw";
+      }
+    else if stdenv.hostPlatform.system == "x86_64-linux" then
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/Archi/default.nix:15]: That is the same as the one above. Did you mean x86_64-darwin?

File: pkgs/tools/misc/Archi/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      throw "Unsupported system"
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/Archi/default.nix:16]: I can test if this builds but you need to enable it.

File: pkgs/tools/misc/Archi/default.nix (hunk 1 above)
Comment (SuperSandro2000) [inline pkgs/tools/misc/Archi/default.nix:30]: Can we use mv here?

File: pkgs/tools/misc/Archi/default.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    description = "ArchiMate modelling toolkit";
+    longDescription = ''
+      Archi is an open source modelling toolkit to create ArchiMate
+      models and sketches.
+    '';
+    homepage = "https://www.archimatetool.com/";
+    license = stdenv.lib.licenses.mit;
+    platforms = stdenv.lib.platforms.linux;
+    maintainers = [ stdenv.lib.maintainers.earldouglas ];
+  };
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/Archi/default.nix:43]:
```suggestion
  meta = with stdenv.lib; {
    description = "ArchiMate modelling toolkit";
    longDescription = ''
      Archi is an open source modelling toolkit to create ArchiMate
      models and sketches.
    '';
    homepage = "https://www.archimatetool.com/";
    license = licenses.mit;
    platforms = platforms.linux;
    maintainers = with maintainers; [ earldouglas ];
  };
```

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -688,6 +688,8 @@ in
     inherit (darwin.apple_sdk.frameworks) Carbon Cocoa Kernel;
   };
 
+  Archi = callPackage ../tools/misc/Archi { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:691]:
```suggestion
  archi = callPackage ../tools/misc/archi { };
```

Names must be lowercased at the beginning.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Darwin patch:
```patch
diff --git a/pkgs/tools/misc/Archi/default.nix b/pkgs/tools/misc/Archi/default.nix
index 137f679b1da0..fa2db41760a8 100644
--- a/pkgs/tools/misc/Archi/default.nix
+++ b/pkgs/tools/misc/Archi/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl }:
+{ stdenv, fetchurl, fetchzip }:
 
 stdenv.mkDerivation rec {
 
@@ -11,22 +11,21 @@ stdenv.mkDerivation rec {
         url = "https://www.archimatetool.com/downloads/archi/Archi-Linux64-${version}.tgz";
         sha256 = "0sd57cfnh5q2p17sd86c8wgmqyipg29rz6iaa5brq8mwn8ps2fdw";
       }
-    else if stdenv.hostPlatform.system == "x86_64-linux" then
-      throw "Unsupported system"
-      /*
-       * A Mac version is available; needs to be tested on a Mac system:
-       * fetchurl {
-       *   url = "https://www.archimatetool.com/downloads/archi/Archi-Mac-${version}.zip";
-       *   sha256 = "0zg0bs9i2kkr11bbssfxhrp4ym2zzhb2bk7m38z9cj99h7pmgyhr";
-       * }
-       */
+    else if stdenv.hostPlatform.system == "x86_64-darwin" then
+      fetchzip {
+        url = "https://www.archimatetool.com/downloads/archi/Archi-Mac-${version}.zip";
+        sha256 = "1h05lal5jnjwm30dbqvd6gisgrmf1an8xf34f01gs9pwqvqfvmxc";
+      }
     else
        throw "Unsupported system";
 
-  installPhase = ''
+  installPhase = if stdenv.hostPlatform.system == "x86_64-darwin" then ''
+    mkdir -p "$out/Applications"
+    mv Archi.app "$out/Applications/"
+  '' else ''
+    install -Dm755 Archi $out/bin/Archi
+    rm Archi
     cp -r . $out
-    mkdir -p $out/bin
-    ln -s $out/Archi $out/bin/Archi
   '';
 
   meta = with stdenv.lib; {
@@ -37,7 +36,7 @@ stdenv.mkDerivation rec {
     '';
     homepage = "https://www.archimatetool.com/";
     license = licenses.mit;
-    platforms = platforms.linux;
+    platforms = platforms.unix;
     maintainers = with maintainers; [ earldouglas ];
   };
 }
```

Also please squash the commits together.

File: pkgs/tools/misc/archi/default.nix [hunk 7]
```diff
@@ -0,0 +1,64 @@
+{ stdenv
+, fetchurl
+, fetchzip
+, autoPatchelfHook
+, libsecret
+}:
+
+stdenv.mkDerivation rec {
+
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/archi/default.nix:9]:
```suggestion
```

File: pkgs/tools/misc/archi/default.nix [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  nativeBuildInputs = [
+    autoPatchelfHook
+  ];
+
+  installPhase =
+    if stdenv.hostPlatform.system == "x86_64-linux" then
+      ''
+        for i in configuration features p2 plugins Archi.ini Archi
+        do
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/archi/default.nix:39]:
```suggestion
        for f in configuration features p2 plugins Archi.ini Archi; do
```

File: pkgs/tools/misc/archi/default.nix [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+          mkdir -p $out/bin
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/archi/default.nix:40]:
```suggestion
```

File: pkgs/tools/misc/archi/default.nix [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    libsecret
+  ];
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+  ];
+
+  installPhase =
+    if stdenv.hostPlatform.system == "x86_64-linux" then
+      ''
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/archi/default.nix:37]:
```suggestion
      ''
      mkdir -p $out/bin
```

File: pkgs/tools/misc/archi/default.nix [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+        for i in configuration features p2 plugins Archi.ini Archi
+        do
+          mkdir -p $out/bin
+          cp -rf $i $out/bin/
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/archi/default.nix:41]:
```suggestion
          cp $f $out/bin/
```

File: pkgs/tools/misc/archi/default.nix [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+        done
+
+        install -D -m755 Archi $out/bin/Archi
+        rm Archi
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/archi/default.nix:45]:
```suggestion
```

File: pkgs/tools/misc/archi/default.nix [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+
+  meta = with stdenv.lib; {
+    description = "ArchiMate modelling toolkit";
+    longDescription = ''
+      Archi is an open source modelling toolkit to create ArchiMate
+      models and sketches.
+    '';
+    homepage = "https://www.archimatetool.com/";
+    license = licenses.mit;
+    platforms = platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/archi/default.nix:61]:
```suggestion
    platforms = platforms.linux ++ platforms.darwin;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103355` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>archi</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103355` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>archi</li>
  </ul>
</details>

**Pull request merged**